File: 0605-can-place-flowers/solution.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        if (n == 0) return true;
        if (n == 1 && flowerbed.size() == 1 && flowerbed[0] == 0) return true;
        int numPlanted = 0;
        for (int i = 0; i < flowerbed.size(); ++i) {
            if (i == 0 && i + 1 < flowerbed.size() && flowerbed[i] == 0 && flowerbed[i + 1] == 0) {
                flowerbed[i] = 1;
                numPlanted++;
            } else if (i != 0 && i != flowerbed.size() - 1 
                        && flowerbed[i] == 0
                        && flowerbed[i - 1] == 0 
                        && flowerbed[i+1] == 0) {
                flowerbed[i] = 1;
                numPlanted++;
            } else if (i == flowerbed.size() - 1 && i - 1 >= 0 && flowerbed[i] == 0 && 
                flowerbed[i - 1] == 0) {
                flowerbed[i] = 1;
                numPlanted++;
            }

            if (numPlanted == n) return true;
        }

        return false;
    }
};

File: 0605-can-place-flowers/solution.py
class Solution(object):
    def canPlaceFlowers(self, flowerbed, n):
        """
        :type flowerbed: List[int]
        :type n: int
        :rtype: bool
        """
        if n == 0:
            return True
        bed = [0] + flowerbed + [0]
        numPlanted = 0
        for i in range(1, len(flowerbed) + 1):
            if bed[i - 1] == bed[i] == bed[i+1] == 0:
                numPlanted += 1
                bed[i] = 1
            
            if numPlanted == n:
                return True

        return False

        
